doc/release-notes.md

```diff
@@ -4,6 +4,56 @@ release-notes at release time)
 Notable changes
 ===============
 
+Fixed bugs in the testnet Orchard circuit
+-----------------------------------------
+
+In the `zcashd v4.5.0` release notes we indicated that a testnet rollback might
+occur to update the consensus rules, if we needed to make backwards-incompatible
+changes. Shortly after `zcashd v4.5.0` was released, during another internal
+review of the Orchard circuit, we identified two bugs that would affect the
+upcoming testnet activation of NU5:
+
+- The diversifier base `g_d_old`, for the note being spent, is required to be a
+  non-identity point. A note created from a payment address with `g_d` set to
+  the identity (via collaboration between sender and recipient) could be spent
+  multiple times with different nullifiers (corresponding to different `ivk`s).
+  The code outside the circuit correctly enforced the non-identity requirement,
+  but the circuit did not correctly constrain this, and allowed the prover to
+  witness the identity.
+
+- SinsemillaCommit can be modeled as a Pedersen commitment to an output of
+  SinsemillaHash: `SinsemillaCommit(r, M) = SinsemillaHashToPoint(M) + [r] R`.
+  The specification used incomplete addition here, matching its use inside
+  SinsemillaHash. However, unlike in SinsemillaHash, an exceptional case can be
+  produced here when `r = 0`. The derivations of `rivk` (for computing `ivk`)
+  and `rcm` (for computing the note commitment) normally ensure that `r = 0`
+  can only occur with negligible probability, but these derivations are not
+  checked by the circuit for efficiency; thus SinsemillaCommit needs to use
+  complete addition.
+
+These bugs do not affect mainnet, as `zcashd v4.5.0` only set the activation
+height for NU5 on testnet for testing purposes. Nevertheless, in the interest of
+keeping the testnet environment as close to mainnet as possible, we are fixing
+these bugs immediately. This means a change to the NU5 consensus rules, and a
+new testnet activation height for NU5.
+
+To this end, the following changes are made in `zcashd v4.5.1`:
+
+- The consensus branch ID for NU5 is changed to `0x37519621`.
+- The protocol version indicating NU5-aware testnet nodes is set to `170015`.
```

Comment:
Do we want to mention the corresponding change to `170017` on mainnet?

Reply:
Nah, the release notes are dense enough as it is, and there's (unfortunately) every possibility that it changes again. People shouldn't rely on this protocol version number until v5.0.0 in any case, so no point drawing more attention to it than needed.